rules/Transform/Rector/Assign/PropertyFetchToMethodCallRector.php

```diff
@@ -172,4 +178,38 @@ private function matchPropertyFetchCandidate(PropertyFetch $propertyFetch): ?Pro
 
         return null;
     }
+
+    private function processPropertyFetch(PropertyFetch $propertyFetch): ?MethodCall
+    {
+        return $this->transformPropertyFetchToMethodCall($propertyFetch);
+    }
+
+    private function transformPropertyFetchToMethodCall(PropertyFetch $propertyFetch): ?MethodCall
+    {
+        $propertyToMethodCall = $this->matchPropertyFetchCandidate($propertyFetch);
+        if (! $propertyToMethodCall instanceof PropertyFetchToMethodCall) {
+            return null;
+        }
+
+        $newMethod = $propertyToMethodCall->getNewGetMethod();
+        if ($newMethod === '') {
+            return null;
+        }
+
+        $args = [];
+
+        if ($propertyToMethodCall->getNewGetArguments() !== []) {
+            $args = $this->nodeFactory->createArgs($propertyToMethodCall->getNewGetArguments());
+        }
+
+        if ($propertyToMethodCall->isEquivalent()) {
+            $newMethod = $this->getName($propertyFetch->name);
+        }
+
+        if ($newMethod === null) {
+            throw new ShouldNotHappenException();
+        }
+
+        return $this->nodeFactory->createMethodCall($propertyFetch->var, $newMethod, $args);
+    }
```

Comment:
The class seems pretty huge now.

Could you extract a service that is would handle the most complex part of this rule? 
That way the rule is more readable and easier to debug for future generations.